tests/rptest/services/utils.py

```diff
@@ -36,3 +45,147 @@ def __str__(self):
+                return True
+        return False
+
+    def _check_memory_leak(self, host):
+        # Special case for LeakSanitizer errors, where tiny leaks
+        # are permitted, as they can occur during Seastar shutdown.
+        # See https://github.com/redpanda-data/redpanda/issues/3626
+        for summary_line in self._capture_log(host,
+                                              "SUMMARY: AddressSanitizer:"):
+            m = re.match(
+                "SUMMARY: AddressSanitizer: (\d+) byte\(s\) leaked in (\d+) allocation\(s\).",
+                summary_line.strip())
+            if m and int(m.group(1)) < 1024:
+                self.logger.warn(
+                    f"Ignoring memory leak, small quantity: {summary_line}")
+                return True
+        return False
+
+    def _check_oversized_allocations(self, line):
+        m = re.search("oversized allocation: (\d+) byte", line)
+        if m and int(m.group(1)) <= self.MAX_ALLOCATION_SIZE:
+            self.logger.warn(
+                f"Ignoring oversized allocation, {m.group(1)} is less than the max allowable allocation size of {self.MAX_ALLOCATION_SIZE} bytes"
+            )
+            return True
+        return False
+
+    def _search(self, nodes):
+        bad_lines = collections.defaultdict(list)
+        test_name = self._context.function_name
+        for node in nodes:
+            hostname = self._get_hostname(node)
+            self.logger.info(f"Scanning node {hostname} log for errors...")
+            # Prepare/Build capture func shortcut
+            # Iterate
+            for line in self._capture_log(node, self.match_expr):
+                line = line.strip()
+                # Check if this line holds error
+                allowed = self._check_if_line_allowed(line)
+                # Check for memory leaks
+                if 'LeakSanitizer' in line:
+                    allowed = self._check_memory_leak(node)
+                # Check for oversized allocations
+                if "oversized allocation" in line:
+                    allowed = self._check_oversized_allocations(line)
+                # If detected bad lines, log it and add to the list
+                if not allowed:
+                    bad_lines[node].append(line)
+                    self.logger.warn(f"[{test_name}] Unexpected log line on "
+                                     f"{hostname}: {line}")
+        return bad_lines
+
+    def search_logs(self, nodes):
+        # Do log search
+        bad_loglines = self._search(nodes)
+        # If anything, raise exception
+        if bad_loglines:
+            raise BadLogLines(bad_loglines)
+
+
+class LogSearchLocal(LogSearch):
+    def __init__(self, test_context, allow_list, logger, targetpath) -> None:
+        super().__init__(test_context, allow_list, logger)
+        self.targetpath = targetpath
+
+    def _capture_log(self, node, expr) -> Generator[str, None, None]:
```

Comment:
It assumes that kubectl command would work. We might want to add an Exception in case it failed to capture logs for pod

Reply:
Agreed. Will do.